src/txn/txn_timestamp.c

```diff
@@ -162,7 +162,15 @@ __txn_global_query_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *tsp, cons
     WT_STAT_CONN_INCR(session, txn_query_ts);
     WT_RET(__wt_config_gets(session, cfg, "get", &cval));
     if (WT_STRING_MATCH("all_durable", cval.str, cval.len)) {
-        ts = txn_global->has_durable_timestamp ? txn_global->durable_timestamp : 0;
+        /*
```

Comment:
is there any reason why this is not under the '__wt_readlock'?

Reply:
That's a good question, the previous check was also outside the readlock so I was just following that logic. I think the readlock is there for walking the concurrent transactions but I'm not sure if it's needed to check if the durable has been set yet. I can't think of any race conditions at the moment.

This line similarly checks the txn_global->has_oldest_timestamp without the txn_global lock. 
https://github.com/wiredtiger/wiredtiger/blob/439cd2fbc72e0a0dcd19e86205feffd5430440c6/src/txn/txn_timestamp.c#L100

Do you think we should move these to inside the lock?